manifests/server.pp

```diff
@@ -78,5 +78,8 @@
     Anchor['mysql::server::end']
   }
 
-
+Class['mysql::server'] -> Mysql_database <||>
+Class['mysql::server'] -> Mysql_user <||>
```

Comment:
Instead of collectors, could you add autorequires for `Class['mysql::server']` to the `mysql_database` and `mysql_user` types? Adding collectors without filters could do something bad for anyone using virtual resources, like creating all databases on all mysql servers.